src/workerd/api/node/zlib-util.c++

```diff
@@ -3,18 +3,20 @@
 //     https://opensource.org/licenses/Apache-2.0
 // Copyright Joyent and Node contributors. All rights reserved. MIT license.
 
-#include "util.h"
 #include "zlib-util.h"
 
+#include "util.h"
+#include <workerd/jsg/exception.h>
+
 #include "nbytes.h"
+#include "workerd/jsg/jsg.h"
 
 // The following implementation is adapted from Node.js
 // and therefore follows Node.js style as opposed to kj style.
 // Latest implementation of Node.js zlib can be found at:
 // https://github.com/nodejs/node/blob/main/src/node_zlib.cc
 namespace workerd::api::node {
-
-kj::ArrayPtr<kj::byte> ZlibUtil::getInputFromSource(InputSource& data) {
+kj::ArrayPtr<kj::byte> getInputFromSource(ZlibUtil::InputSource& data) {
```

Comment:
You can keep it as it is, and just move it while passing to an argument:

```suggestion
kj::ArrayPtr<kj::byte> getInputFromSource(ZlibUtil::InputSource data) {
```

Reply:
This is to keep it out of the header since it's private. Speeds compiles a little.

Reply:
Then I recommend making it const

Reply:
I'm not sure that I follow. It's not a member function so it can't be const. It's already static but I can add the keyword.

Reply:
I mean:

```suggestion
kj::ArrayPtr<kj::byte> getInputFromSource(const ZlibUtil::InputSource& data) {
```

Reply:
Ah and I guess that means the return type should be kj::ArrayPtr<const kj::byte> though I need to double check with all these funky templates